File: SearchGSTIN.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Net
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports TaxProGST.API
Imports TaxProGST.JsonModels
Imports Newtonsoft.Json
Imports System.IO

Public Class frmSearchGSTIN
    Private Sub frmSearchGSTIN_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12
    End Sub
    Private Async Sub BtnSearch_Click(sender As Object, e As EventArgs) Handles BtnSearch.Click
        Dim fileName = Application.StartupPath + "\" + TxtSearchGSTIN.Text.ToString() + ".dat"
        Dim txnResp As TxnRespWithObj(Of SearchTaxpayerJson) = Await PublicAPI.SearchTaxPayerAsync("1613471119", "Dhanchhaya@123", "24AAGCA8671A1ZX", TxtSearchGSTIN.Text.ToString())
        If (txnResp.IsSuccess) Then
            'MsgBox(JsonConvert.SerializeObject(txnResp.RespObj))
            TxtGSTINDet.Text = Newtonsoft.Json.JsonConvert.SerializeObject(txnResp.RespObj)
            Dim fs As FileStream = New FileStream(fileName, FileMode.OpenOrCreate, FileAccess.ReadWrite)
            Dim s As StreamWriter = New StreamWriter(fs)
            s.Close()
            fs.Close()
            Dim fs1 As FileStream = New FileStream(fileName, FileMode.Append, FileAccess.Write)
            Dim s1 As StreamWriter = New StreamWriter(fs1)
            s1.Write(Newtonsoft.Json.JsonConvert.SerializeObject(txnResp.RespObj))
            s1.Close()
            fs1.Close()
            MsgBox("GSTIN detail fetched successfully.", MsgBoxStyle.OkOnly)
        Else
            MsgBox(txnResp.TxnOutcome)
            TxtGSTINDet.Text = txnResp.TxnOutcome
        End If
    End Sub
End Class

File: SearchGSTIN.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.IO;
using System.Linq;
using System.Net;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using TaxProEInvoice.API;

namespace EWayBillTool
{
    public partial class SearchGSTIN : Form
    {
        private AppMain Appmain = new AppMain();
        public SearchGSTIN()
        {
            InitializeComponent();
        }
        private void frmSearchGSTIN_Load(object sender, EventArgs e)
        {
            ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
        }

        private async void BtnSearch_Click(object sender, EventArgs e)
        {
            var fileName = Application.StartupPath + @"\" + TxtSearchGSTIN.Text.ToString() + ".dat";            
            TaxProEInvoice.API.TxnRespWithObj<RespPlGetGSTIN> txnResp = await eInvoiceAPI.GetGSTINDetailsAsync(Appmain.eInvSession, TxtSearchGSTIN.Text.ToString());
            if (txnResp.IsSuccess)
            {
                TxtGSTINDet.Text = Newtonsoft.Json.JsonConvert.SerializeObject(txnResp.RespObj);
                FileStream fs = new FileStream(fileName, FileMode.OpenOrCreate, FileAccess.ReadWrite);
                StreamWriter s = new StreamWriter(fs);
                s.Close();
                fs.Close();
                FileStream fs1 = new FileStream(fileName, FileMode.Append, FileAccess.Write);
                StreamWriter s1 = new StreamWriter(fs1);
                s1.Write(Newtonsoft.Json.JsonConvert.SerializeObject(txnResp.RespObj));
                s1.Close();
                fs1.Close();                
                MessageBox.Show("GSTIN detail fetched successfully.","", MessageBoxButtons.OK);
            }
            else
            {
                MessageBox.Show(txnResp.TxnOutcome);
                TxtGSTINDet.Text = txnResp.TxnOutcome;
            }
        }
    }
}
